_posts/2020-10-22-how-to-update-to-aws-provider-v3.adoc

```diff
@@ -0,0 +1,191 @@
+---
+title: How to update Gruntwork references to Version 3 of the Terraform AWS provider
+categories: Foundations
+image: /assets/img/guides/infrastructure-as-code-library/grunty-grid.png
+excerpt: Learn how to update the Gruntwork Reference Architecture and your infrastructure code that depends on the Gruntwork Infrastructure as Code Library to use the latest version of the Terraform AWS provider.
+tags: ["aws", "terraform", "terragrunt"]
+cloud: ["aws"]
+redirect_from: /static/guides/foundations/how-to-update-to-aws-provider-v3/
+---
+:page-type: guide
+:page-layout: post
+
+:toc:
+:toc-placement!:
+
+// GitHub specific settings. See https://gist.github.com/dcode/0cfbf2699a1fe9b46ff04c41721dda74 for details.
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+toc::[]
+endif::[]
+
+== Intro
+
+This guide will walk you through how to update the https://gruntwork.io/reference-architecture/[Gruntwork Reference
+Architecture] and any code that depends on the
+https://gruntwork.io/infrastructure-as-code-library/[Gruntwork Infrastructure as Code Library] to the latest version
+(v3) of the Terraform AWS provider.
+
+=== What you'll learn in this guide
+
+This guide consists of three main sections:
+
+<<core_concepts>>::
+  An overview of version 3 of the AWS provider and why it is important to update your code for compatibility.
+
+<<deployment_walkthrough>>::
+  The steps you need to take to update your code relying on the Gruntwork Infrastructure as Code library and your
+  version of the Gruntwork Reference Architecture with compatibility with AWS provider v3.
+
+
+[[core_concepts]]
+== Core Concepts
+
+=== Background
+
+Version https://github.com/terraform-providers/terraform-provider-aws/releases/tag/v3.0.0[v3.0.0] of the Terraform AWS
+provider was released on July 30th 2020 with backwards incompatible updates. Following the provider release cycle,
+future releases from this point onward would only be compatible with 3.X. This means that if you wish to use any new
+resources or data sources that are released from this point onwards, you will need to be compatible with version 3 of
+the AWS provider to be able to pull those changes in.
+
+Since this is a major version change, there are a number of backwards incompatible changes that were introduced. In
+order to be able to use version 3, you need to update the resources and data sources with backwards incompatible
+changes.
```

Comment:
NIT: it's not so much the resources / data sources you need to update, but _your code_ and _your Terraform state file_.

Reply:
NOTE: There are no terraform state changes in this update if it is just the aws provider changes (there is a bunch in the ref arch because we are jumping 5-10 backwards incompatible changes from our modules).

I'll rephrase this, but I was really going for "update your resource and data source blocks".

Reply:
Rephrased in e198cac59996c0af5e0f57978b119abbb95b7c4e